WordPress/Classes/ViewRelated/Reader/ReaderPostDetailViewController.m

```diff
@@ -265,6 +265,32 @@ - (void)setupWithPostID:(NSNumber *)postID siteID:(NSNumber *)siteID
 
 #pragma mark - Accessor methods
 
+- (void)setPost:(ReaderPost *)post
+{
+    if (post == _post) {
+        return;
+    }
+
+    if (!post) {
+        _post = nil;
+        return;
+    }
+
+    _post = [self postInContext:post.objectID];
```

Comment:
Well, `ReaderPostsViewController` is going away, replaced by `ReaderStreamViewController`.  However, in retrospect, since the ReaderStreamViewController is the oddball using a child context it probably makes more sense for it to get the post from the main context before passing to either the detail or the comment vc. That will at least keep all the weirdness in one class.